src/coreclr/nativeaot/Runtime/unix/PalRedhawkInline.h

```diff
@@ -94,16 +94,20 @@ FORCEINLINE int64_t PalInterlockedCompareExchange64(_Inout_ int64_t volatile *pD
     return result;
 }
 
-#if defined(HOST_AMD64) || defined(HOST_ARM64)
+#if defined(HOST_AMD64) || defined(HOST_ARM64) || defined(HOST_LOONGARCH64)
 FORCEINLINE uint8_t PalInterlockedCompareExchange128(_Inout_ int64_t volatile *pDst, int64_t iValueHigh, int64_t iValueLow, int64_t *pComparandAndResult)
 {
     __int128_t iComparand = ((__int128_t)pComparandAndResult[1] << 64) + (uint64_t)pComparandAndResult[0];
+    // NOTE: for LoongArch64, it supports 128bits atomic from 3A6000-CPU which is ISA1.1's version.
+    // The LA64's compiler will translate the `__sync_val_compare_and_swap` into calling the libatomic's library interface to emulate
```

Comment:
Does the build target ISA1.0 or ISA1.1 by default? If it targets ISA1.0 by default, this implementation is broken. This should be TODO instead of NOTE, ideally with a link to an issue that tracks fixing it.

Reply:
Changed to TODO linking https://github.com/dotnet/runtime/issues/109276. This is @shushanhf's comment. CI cross build is not linking libatomic, so it should be ISA1.1.